index.js

```diff
@@ -1,18 +1,110 @@
-const readline = require('readline');
+const readline = require("readline");
 const rl = readline.createInterface(process.stdin, process.stdout);
+// Valid string inputs -----------------------------
+const correctAnswer = "yes" || "y" || "ys";
+const wrongAnswer = "no" || "n";
+const highAnswer = "higher" || "h" || "high";
+const playComputerGuessing = 'computer';
+const playPlayerGuessing = 'player'
 
+// Function list -----------------------------------
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     rl.question(questionText, resolve);
   });
+} // Sanatizes string input
+function sanitizeString(string) {
+  string = string
+    .toString()
+    .trim()
+    .toLowerCase();
+  return string;
+} // Creates a random integer between two numbers
+function randomInteger(min, max) {
+  return Math.floor(Math.random() * (max - min)) + min;
 }
 
-start();
+chooseGameType();
+
+async function chooseGameType () {
+console.log('\nPlease type either computer, to have the computer guess the number \nor player for the player to guess the number.\n');
+
+let chooseType = await ask('Would you like to guess the number or have me (computer) guess the number? ')
+  if (chooseType === playComputerGuessing) {
+    start();
+  } else if (chooseType === playPlayerGuessing) {
+    reverseStart();
+  }
+}
+
+//start();
 
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
+  console.log(
+    "\nLet's play a game where you (human) make up a number\nand I (computer) try to guess it.\n"
+  );
+  // Game setup
+  let setRangeMin = await ask("What is the minimum number I can guess to? ");
+  setRangeMin = +setRangeMin; // changes setRangeMin to a number
+  let setRangeMax = await ask("What is the maximum number I can guess to? ");
+  setRangeMax = +setRangeMax; // changes setRangeMax to a number
+  const secretNumber = await ask("What is your secret number? ");
+  let computerGuess = randomInteger(setRangeMin, setRangeMax);
+  let guessQuestion = null;
+
+  while (guessQuestion !== correctAnswer) {
+    guessQuestion = await ask(`Is your number ${computerGuess}? `);
+
+    if (guessQuestion === sanitizeString(wrongAnswer)) {
```

Comment:
Here you could use the array method described above to correctly check if the user enters one of a list of words
```js
wrongAnswers = ['no', 'n']
if (wrongAnswers.includes(sanitizeString(guessQuestion)) {
// do things
}
```